pkg/template/template.go

```diff
@@ -38,9 +39,16 @@ func New(w io.Writer, width int, colorEnabled bool) Template {
 		output:       w,
 		tp:           tableprinter.New(w, true, width),
 		width:        width,
+		funcs:        template.FuncMap{},
 	}
 }
 
+// RegisterFunc registers a named function or overwrites a built-in function.
+// Call this before Parse.
+func (t *Template) RegisterFunc(name string, f func(fields ...interface{}) (string, error)) {
```

Comment:
Only question I have is should we change this to be `Funcs` and have the argument be a `template.FuncMap` to match the `text/template` package? My vote would be yes to keep consistency with the `text/template` package since we have been doing that with `Parse` and `Execute`. I know I was the one who suggested `RegisterFunc` in the first place 😞

Reply:
I've been trying to think of something along these lines as well. Having written an example for testing, the user ends up having to do a lot of what `text/template` does internally.

Do we support devs calling `RegisterFuncs` multiple times and just merge the maps? Last wins, which would be consistent with possibly overriding defaults?

Reply:
That is what I was thinking. We support calling `Funcs` multiple times and just merging the maps where the latest entry for a key wins. That is how `Funcs` works in `text/template`.

Reply:
@heaths I went ahead and made the changes I suggested. Let me know what you think.